### PR TITLE
Adds vkernel feature, fix for copying Kernel PTEs to booter's pgtbl, feedback cleanups

### PR DESCRIPTION
**Pull request by phanikishoreg**
### Summary of this PR

This PR is a feedback fix for the PR:https://github.com/gparmer/Composite/pull/191, which adds vkernel as a major feature. 

### Code Quality

As part of this pull request, I've considered the following:

Style:

- [x] Comments adhere to the Style Guide (SG)
- [x] Spacing adhere's to the SG 
- [x] Naming adhere's to the SG
- [x] All other aspects of the SG are adhered to, or exceptions are justified in this pull request

Code Craftsmanship:

- [x] I've made an attempt to remove all redundant code
- [x] I've considered ways in which my changes might impact existing code, and cleaned it up
- [x] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [x] I've commented appropriately where code is tricky
- [x] I agree that there is no "throw-away" code, and that code in this PR is of high quality: *any such hackish code is well commented with `TODO`/`FIXME`s*

### CODE REVIEW

File: src/components/lib/cos_kernel_api.c [hunk 1]
```diff
@@ -689,11 +812,29 @@ cos_hw_cycles_per_usec(hwcap_t hwc)
 { return call_cap_op(hwc, CAPTBL_OP_HW_CYC_USEC, 0, 0, 0, 0); }
 
 void *
-cos_hw_map(struct cos_compinfo *ci, hwcap_t hwc, paddr_t pa)
+cos_hw_map(struct cos_compinfo *ci, hwcap_t hwc, paddr_t pa, unsigned int len)
 {
-	vaddr_t va = __page_bump_valloc(ci);
+	size_t sz;
+	vaddr_t fva, va;
```
Comment (gparmer) [inline src/components/lib/cos_kernel_api.c]: unfortunately, I can't figure out what `f` stands for here.

Comment (phanikishoreg) [inline src/components/lib/cos_kernel_api.c]: Yes. Okay, I'll do that.

File: src/kernel/include/shared/cos_types.h [hunk 2]
```diff
@@ -259,8 +259,22 @@ enum {
...
-	BOOT_MEM_KM_BASE = PAGE_SIZE, /* kernel memory @ first page, not at address 0 to avoid NULL */
-	BOOT_MEM_PM_BASE = 0x80000000, /* user memory @ 2 GB */
+	BOOT_MEM_KM_BASE = PGD_SIZE, /* kernel & user memory @ 4M, pgd aligned start address */
+};
+
+enum {
+	/* thread register states */
+	THD_GET_IP,
+	THD_GET_SP,
+	THD_GET_BP,
```
Comment (gparmer) [inline src/kernel/include/shared/cos_types.h]: we should likely try and make these generic across architectures.

BP = FP (frame pointer)
IP = IP
SP = SP
AX = 1
BX = 2
CX = 3
DX = 4
SI = 5
DI = 6

(as in `THD_GET_1`)

Another question is if we want these are generic register identifiers, or if we want the action `GET` associated with them.  This is fine for now, but we might change them in the future.

**Comment by gparmer** [comment]
This is in good enough of a state to merge so that @ryuxin  can start using it.  Keep in mind the comments I made...future PR.

**Pull request merged**